### PR TITLE
add memray package

### PR DESCRIPTION
**Pull request by matthiasdiener**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
This is ready for review @conda-forge/help-python @conda-forge/help-python-c

File: recipes/memray/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - rich <11.0.0
+    - jinja2
+
+test:
+  imports:
+    - memray
+  requires:
+    - pip
+  commands:
+    - pip check
```
Comment (dbast) [inline recipes/memray/meta.yaml:38]: Also testing that the cli entry point is there and at least showing the help works.
```suggestion
    - pip check
    - memray -h
```

File: recipes/memray/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    - make
+  host:
+    - python
+    - pip
+    - libunwind
+    - cython
+  run:
+    - python
+    - rich <11.0.0
+    - jinja2
```
Comment (dbast) [inline recipes/memray/meta.yaml:30]: According to https://github.com/bloomberg/memray/blob/main/setup.py#L85 also
```suggestion
    - jinja2
    - typing-extensions  # [py<38]
```

**Comment by matthiasdiener** [comment]
Thank you for the suggestions @dbast ! This is ready for another review.

**Review by dbast — APPROVED** [review]

File: recipes/memray/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  description: |
+    Memray is a memory profiler for Python. It can track memory allocations in
+    Python code, in native extension modules, and in the Python interpreter
+    itself. It can generate several different types of reports to help you
+    analyze the captured memory usage data. While commonly used as a CLI tool,
+    it can also be used as a library to perform more fine-grained profiling
+    tasks.
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE
```
Comment (dbast) [inline recipes/memray/meta.yaml:54]: They vendor the libbacktrace which then also needs to be represented here as it seems to be not present in the root LICENSE file see https://github.com/bloomberg/memray/blob/main/src/vendor/libbacktrace/LICENSE
```suggestion
  license: Apache-2.0 + BSD-3-Clause
  license_family: Apache
  license_file: 
    - LICENSE
    - src/libbacktrace/LICENSE
```

**Comment by dbast** [comment]
@matthiasdiener Thanks!

**Comment by matthiasdiener** [comment]
Thank you @dbast!

**Pull request merged**